### PR TITLE
Cleanup among PCI/Memory constants

### PR DESCRIPTION
**Pull request by FeralChild64**
When experimenting with PCI code I have found that start/end addresses of various memory areas are scattered across various files. This is dangerous - one can easily put a new PCI card in an already occupied space. It is better to have all these constants in one place.

Additionally, I have removed one empty header file.

### CODE REVIEW

File: include/pci_bus.h [hunk 1]
```diff
@@ -24,6 +24,16 @@
 
 #include "mem.h"
 
+// Start of PCI address space
+constexpr uint32_t PCI_MEMORY_BASE      = 0xc000'0000; // gives 3072 MB max RAM
```
Comment (johnnovak) [inline include/pci_bus.h:28]: UpperCamelCase for public constants please.

File: src/hardware/vga_s3.cpp [hunk 2]
```diff
@@ -902,24 +902,19 @@ struct PCI_VGADevice : public PCI_Device {
...
-
-		uint32_t gfx_address_space_mmio = (((uint32_t)S3_LFB_BASE + 0x1000000) &
-		                                   0xfffffff0); // memory space,
-		                                                // within first 4GB
-		registers[0x14] = (uint8_t)(gfx_address_space_mmio & 0xff); // base addres 0
-		registers[0x15] = (uint8_t)((gfx_address_space_mmio >> 8) & 0xff);
-		registers[0x16] = (uint8_t)((gfx_address_space_mmio >> 16) & 0xff);
-		registers[0x17] = (uint8_t)((gfx_address_space_mmio >> 24) & 0xff);
+		// BAR0 - memory space, within first 4GB
+		static_assert((PCI_GFX_LFB_BASE & 0xf) == 0);
```
Comment (johnnovak) [inline src/hardware/vga_s3.cpp:906]: Maybe a commenting that you're asserting 16-byte alignment here.

File: src/ints/int10_modes.cpp [hunk 3]
```diff
@@ -1769,11 +1770,14 @@ bool INT10_SetVideoMode(uint16_t mode)
 		IO_Write(crtc_base+1,0);
 		//  Setup the linear frame buffer
 		IO_Write(crtc_base,0x59);
-		IO_Write(crtc_base+1,(uint8_t)((S3_LFB_BASE >> 24)&0xff));
+		IO_Write(crtc_base+1,
+		         static_cast<uint8_t>((PCI_GFX_LFB_BASE >> 24) & 0xff));
```
Comment (johnnovak) [inline src/ints/int10_modes.cpp:1774]: Minor: reformat

**Review by johnnovak — APPROVED** [review]
Looking good @FeralChild64 , just some minor comments.

File: include/pci_bus.h (hunk 1 above)
Comment (FeralChild64) [reply include/pci_bus.h]: OK, changed.

File: src/hardware/vga_s3.cpp (hunk 2 above)
Comment (FeralChild64) [reply src/hardware/vga_s3.cpp]: OK, added a comment - in every such place.

File: src/ints/int10_modes.cpp (hunk 3 above)
Comment (FeralChild64) [reply src/ints/int10_modes.cpp]: OK, used the script.

**Pull request merged**